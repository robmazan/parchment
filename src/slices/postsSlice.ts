import { createSlice, PayloadAction } from "@reduxjs/redux-toolkit";

export interface ReceivePostsParam {
  posts: string[];
}

export interface PostsState {
  posts: string[];
}

const initialState: PostsState = {
  posts: []
};

const postsSlice = createSlice({
  name: "posts",
  initialState,
  reducers: {
    receivePosts: (state, action: PayloadAction<ReceivePostsParam>) => {
      state.posts = [...action.payload.posts];
    }
  }
});

export default postsSlice;
