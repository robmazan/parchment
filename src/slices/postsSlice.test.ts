import postsSlice, { Post } from "../slices/postsSlice";
import deepFreeze from "deep-freeze";

const sampleData: Post[] = require("./__mocks__/samplePosts.json");

const getInitialState = (): Post[] => {
  return deepFreeze([] as Post[]) as Post[];
};

describe("posts slice", () => {
  describe("receivePosts action", () => {
    it("should create an action to receive posts", () => {
      const actionCreator = postsSlice.actions.receivePosts;
      const expectedAction = {
        type: actionCreator.toString(),
        payload: sampleData
      };
      expect(actionCreator(sampleData)).toEqual(expectedAction);
    });
  });

  describe("combined reducer", () => {
    it("should return the initial state", () => {
      expect(getInitialState()).toEqual([] as Post[]);
    });

    it("should handle receivePosts action", () => {
      const reducer = postsSlice.reducer;
      const actionCreator = postsSlice.actions.receivePosts;
      expect(reducer(getInitialState(), actionCreator(sampleData))).toEqual(
        sampleData
      );
    });
  });
});
