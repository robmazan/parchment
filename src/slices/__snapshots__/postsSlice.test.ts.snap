// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts slice receivePosts action should create an action to receive posts 1`] = `
Object {
  "payload": Array [
    Object {
      "authorId": 1,
      "categoryId": 1,
      "content": "Content 1",
      "excerpt": "Excerpt 1",
      "id": 1,
      "title": "Title 1",
    },
    Object {
      "authorId": 2,
      "categoryId": 2,
      "content": "Content 2",
      "excerpt": "Excerpt 2",
      "id": 2,
      "title": "Title 2",
    },
    Object {
      "authorId": 3,
      "categoryId": 3,
      "content": "Content 3",
      "excerpt": "Excerpt 3",
      "id": 3,
      "title": "Title 3",
    },
  ],
  "type": "posts/receivePosts",
}
`;
